src/clients/cli.rs

```diff
@@ -126,51 +123,51 @@ impl Client {
         }
     }
 
-    fn build_run_command<'a, I: Image>(
-        image: &I,
-        command: &'a mut Command,
-        run_args: &RunArgs,
-    ) -> &'a mut Command {
+    fn build_run_command<I: Image>(runnable: &RunnableImage<I>, mut command: Command) -> Command {
         command.arg("run");
 
-        if let Some(network) = run_args.network() {
+        if let Some(network) = runnable.network() {
             command.arg(format!("--network={}", network));
         }
 
-        if let Some(name) = run_args.name() {
+        if let Some(name) = runnable.name() {
             command.arg(format!("--name={}", name));
         }
 
-        for (key, value) in image.env_vars() {
+        let env_vars = runnable.image().env_vars().chain(runnable.env_vars());
+        for (key, value) in env_vars {
             command.arg("-e").arg(format!("{}={}", key, value));
         }
 
-        for (orig, dest) in image.volumes() {
+        let volumes = runnable.image().volumes().chain(runnable.volumes());
+        for (orig, dest) in volumes {
             command.arg("-v").arg(format!("{}:{}", orig, dest));
         }
 
-        if let Some(entrypoint) = image.entrypoint() {
+        if let Some(entrypoint) = runnable.image().entrypoint() {
             command.arg("--entrypoint").arg(entrypoint);
         }
 
-        if let Some(ports) = run_args.ports() {
-            for port in &ports {
+        if let Some(ports) = runnable.ports() {
+            for port in ports {
                 command
                     .arg("-p")
                     .arg(format!("{}:{}", port.local, port.internal));
             }
         } else {
-            for port in image.expose_ports() {
+            for port in runnable.image().expose_ports() {
                 command.arg(format!("--expose={}", port));
             }
             command.arg("-P"); // publish all exposed ports
         }
 
         command
             .arg("-d") // Always run detached
-            .arg(image.descriptor())
-            .args(image.args())
-            .stdout(Stdio::piped())
+            .arg(runnable.image().descriptor())
+            .args(runnable.image_args().clone().into_iter())
```

Comment:
Can we expose forwarding functions on `RunnableImage` instead of reaching into `image()` itself?

We will have to do this for all clients (currently only two), so making this reasonably ergonomic will bring a benefit.